pkg/runtime/core/trainingruntime.go

```diff
@@ -187,6 +187,25 @@ func (r *TrainingRuntime) mergePodSpecOverrides(trainJob *trainer.TrainJob, jobS
 			}) {
 				continue
 			}
+
+			if len(podSpecOverride.Labels) > 0 {
+				if jobSetTemplateSpec.Spec.ReplicatedJobs[i].Template.Spec.Template.Labels == nil {
+					jobSetTemplateSpec.Spec.ReplicatedJobs[i].Template.Spec.Template.Labels = make(map[string]string)
+				}
+				for k, v := range podSpecOverride.Labels {
+					jobSetTemplateSpec.Spec.ReplicatedJobs[i].Template.Spec.Template.Labels[k] = v
+				}
+			}
+
+			if len(podSpecOverride.Annotations) > 0 {
+				if jobSetTemplateSpec.Spec.ReplicatedJobs[i].Template.Spec.Template.Annotations == nil {
+					jobSetTemplateSpec.Spec.ReplicatedJobs[i].Template.Spec.Template.Annotations = make(map[string]string)
+				}
+				for k, v := range podSpecOverride.Annotations {
+					jobSetTemplateSpec.Spec.ReplicatedJobs[i].Template.Spec.Template.Annotations[k] = v
+				}
+			}
+
```

Comment:
Is it really required? My understanding is that those new fields should already be merged in the JobSet template with
https://github.com/kubeflow/trainer/blob/2283173b9a4a160e5f28b768a5ea0d1e2766baeb/pkg/runtime/core/trainingruntime.go#L133

Reply:
This is required here.

1. A Pod has two parts: metadata (which stores labels/annotations) and spec (containers, volumes, etc.).
2. The current merge only targets corev1.PodSpec (i.e., .spec) and does not include metadata, so labels/annotations will not be merged automatically.
3. Therefore, the Labels/Annotations from PodSpecOverrides need to be manually written back into PodTemplateSpec.ObjectMeta.

Reply:
Ah got it. Maybe we can "extend" `PodSpecOverride` to be considered a subset of `PodTemplateSpec` and directly set `jobSetTemplateSpec.Spec.ReplicatedJobs[i].Template.Spec.Template`.

Reply:
@andreyvelich Building the appropriate PodTemplate from PodSpecOverrides has been completed.